### DISCUSSION TITLE
`response.atomic` how does it work, when should we use it?

### DISCUSSION THREAD

**Discussion by mikevercoelen** [kartikk221/hyper-express#222] [Q&A] [answered]
Question about `response.atomic` how does it work exactly? Apologies but I'm a bit confused by the docs that say:

> // We wrap multiple response network calls in atomic callback for best performance

**Comment by kartikk221** [comment, marked as answer]
Hello, the `Response.atomic` method is simply an alias of the `uWS.HttpResponse.cork()` method. 

From what I understand, the purpose of this method is to let `uWebsockets.js` know that you intend to perform multiple Network operations such as write HTTP status, headers, body chunks etc etc in the Response. It is essentially a way to optimize and I think even required in the recent updates when doing async work so `uWebsockets.js` can batch these network write operations as much as possible especially for SSL.

HyperExpress automatically makes use of corking / this method internally wherever needed so you don't have to actually worry about using this method in any way unless you are doing something very low level and directly interacting with `uWebsockets.js` and need to cork.
